src/filters.rs

```diff
@@ -60,6 +60,46 @@ pub fn upcase(input: &Value, _args: &[Value]) -> FilterResult {
     }
 }
 
+
+pub fn downcase(input: &Value, _args: &[Value]) -> FilterResult {
+    match *input {
+        Str(ref s) => Ok(Str(s.to_lowercase())),
+        _ => Err(InvalidType("String expected".to_owned())),
+    }
+}
+
+
+pub fn capitalize(input: &Value, _args: &[Value]) -> FilterResult {
+    match *input {
+        Str(ref s) => Ok(Str(s.char_indices().fold(String::new(), |word, (_, chr)| {
+            let next_char = match word.chars().last() {
+                Some(last) =>
+                    if last.is_whitespace() {
+                        chr.to_uppercase().next().unwrap()
+                    } else {
+                        chr
+                    },
+                _ => chr.to_uppercase().next().unwrap(),
+            }.to_string();
+            word + &next_char
```

Comment:
Re ß:
ß is the **only letter** that doesn't have any uppercase at all. The work-a-round in technical sense is to often replace it with "SS", which is technically two characters, however that never effects us, as ß also can't be used as the beginning of any word as it should only be used to extend the previous vowel. So, while in a uppercasing of a text this would be relevant (and actually mean your lowercased text is shorter than the uppercased one), in a title-case scenario (the one we are looking at), this isn't an issue in practice. Given, that someone could still feed the ß at the beginning of the word, but as far as I understand this code, this would just lead to the usage of "S" instead of "SS".